### PR TITLE
Rename telemetry names and partially implementation on Eclipse Plugin

### PR DESCRIPTION
**Pull request by juniwang**
Changes include:
- client telemetry implementation for eclipse
  - Dialogs/Pages
  - Actions/handlers on Azure explorer
  - tools menu/main memu/popup menu
- trace every rest call to Azure API
- rename telemetry eventnames about the plugin event and telemetry preference event

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by vscjavaci** [comment]
add to whitelist

**Comment by voidfunction** [comment]
Can one of the admins verify this patch?

**Comment by juniwang** [comment]
Build Error "[ERROR] Internal error: java.lang.RuntimeException: org.eclipse.equinox.p2.core.ProvisionException: Unable to read repository at http://download.eclipse.org/releases/luna. Unable to read repository at http://download.eclipse.org/technology/epp/packages/luna/artifacts.jar. connect timed out -> [Help 1]"
Something wrong with the build agent?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateBlobContainer.java [hunk 1]
```diff
@@ -31,7 +32,7 @@ public void actionPerformed(NodeActionEvent e) {
     		connectionString = StorageClientSDKManager.getConnectionString(((StorageNode) parent).getStorageAccount());
     	}
         CreateBlobContainerForm form = new CreateBlobContainerForm(PluginUtil.getParentShell(), connectionString);
-
+        if(parent instanceof StorageNode) { form.setSubscription(new SubscriptionDetail(((StorageNode)parent).getSubscriptionId(), null, null, true)); }
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateBlobContainer.java]: should check null of parent?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java [hunk 2]
```diff
@@ -109,6 +119,78 @@ public void createPartControl(Composite composite) {
...
+				boolean optionDialog = DefaultLoader.getUIHelper().showConfirmation(
+                        "Are you sure you want to clear the queue \"" + queue.getName() + "\"?",
+                        "Service explorer",
+                        new String[]{"Yes", "No"},
+                        null);
+
+                if (optionDialog) {
+                    DefaultLoader.getIdeHelper().runInBackground(null, "Clearing queue messages", false, true, "Clearing queue messages", new Runnable() {
+                        public void run() {
+                            /*try {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java]: Nothing to do here?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java [hunk 3]
```diff
@@ -249,19 +298,8 @@ public void menuAboutToShow(IMenuManager manager) {
...
-                    };
-                    manager.add(action);
-                    action = new Action("Dequeue") {
-                        @Override
-                        public void run() {
-                            dequeueFirstMessage();
-                        }
-                    };
+                    manager.add(fileEditorVirtualNode.createPopupAction(OPEN));
+                    Action action = fileEditorVirtualNode.createPopupAction(DEQUEUE);
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java]: check the null of fileEditorVirtualNode?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java [hunk 4]
```diff
@@ -133,18 +134,20 @@ public boolean performOk() {
 					String oldPrefVal = DataOperations.getProperty(dataFile, Messages.prefVal);
 					DataOperations.updatePropertyValue(doc, Messages.prefVal,
 							String.valueOf(btnPreference.getSelection()));
-					String version = DataOperations.getProperty(dataFile, Messages.version);
+					
+					final String version = DataOperations.getProperty(dataFile, Messages.version);
+					final String newVersion = Activator.getDefault().getBundle().getVersion().toString();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java]: null check needed?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.docker/src/com/microsoft/azuretools/docker/ui/wizards/createhost/AzureNewDockerConfigPage.java [hunk 5]
```diff
@@ -412,6 +413,7 @@ public void selectionChanged(SelectionChangedEvent event) {
 		if (dockerManager.getSubscriptionsList() != null && dockerManager.getSubscriptionsList().size() > 0) {
 			dockerSubscriptionCombo.select(0);			
 			dockerSubscriptionIdTextField.setText(((AzureDockerSubscription) ((StructuredSelection) dockerSubscriptionComboViewer.getSelection()).getFirstElement()).id);
+			wizard.setSubscription((AzureDockerSubscription) ((StructuredSelection) dockerSubscriptionComboViewer.getSelection()).getFirstElement());
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.docker/src/com/microsoft/azuretools/docker/ui/wizards/createhost/AzureNewDockerConfigPage.java]: type check null check needed?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateBlobContainer.java (hunk 1 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateBlobContainer.java]: unneccessary to do null check. `obj instanceof SomeClass` is `false` if `obj` is `null`.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java (hunk 2 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java]: Same behavior as the original version where the inner codes were commented. I don't know why but just keep the same.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java (hunk 3 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java]: `fileEditorVirtualNode` should NEVER be null. It's instantiated in the `QueueFileEditor`'s contructor.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java (hunk 4 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java]: Not sure of this. Extract the original version to a variable only. Let me try to understand the code.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.docker/src/com/microsoft/azuretools/docker/ui/wizards/createhost/AzureNewDockerConfigPage.java (hunk 5 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.docker/src/com/microsoft/azuretools/docker/ui/wizards/createhost/AzureNewDockerConfigPage.java]: According to line 415, it's not needed. @zackliu please take a look at this.

Comment (zackliu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.docker/src/com/microsoft/azuretools/docker/ui/wizards/createhost/AzureNewDockerConfigPage.java]: Yes, it won't be null

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java (hunk 4 above)
Comment (juniwang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/WindowsAzurePreferencePage.java]: Both `getBundle()` and `getVersion` won't be `null` according http://www.docjar.com/html/api/org/eclipse/core/runtime/Plugin.java.html and https://osgi.org/javadoc/r5/core/org/osgi/framework/Bundle.html#getVersion().

**Pull request merged**